Prepare 1.3.0 release

raynelfss opened the pull request:
- Move loose release notes.

<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
-->

### Summary
The following commits set the version number on `stable/1.3` to `1.3.0` in preparation for the `1.3.0` final release.


### Details and comments
The following commits will:
- [x] Set the version number to `1.3.0`.
- [x] Consolidate any loose release notes into the `releasenotes/notes/1.3/` folder.
- [x] Update the `cargo` dependencies.
- [x] Fix any typos, broken links, and bad references in the folder `releasenotes/notes/1.3/`.
- [x] Include a prelude release note for the release.

raynelfss committed:
Initial: Change version to 3.0
- Move loose release notes.

raynelfss committed:
Update cargo dependencies in `Cargo.lock`

raynelfss committed:
1st round (to be rebased)

beckykd commented on releasenotes/notes/1.3/add-mcx-plugins-85e5b248692a36db.yaml:
```diff
@@ -17,7 +17,7 @@ features_transpiler:
-    As well:
+    As well as:
 
     * :class:`.MCXSynthesisDefault`, choosing the most efficient synthesis
```
```suggestion
    * :class:`.MCXSynthesisDefault`, for choosing the most efficient synthesis
```

beckykd commented on releasenotes/notes/1.3/add-more-sabre-trials-9b421f05d2f48d18.yaml:
```diff
@@ -2,10 +2,11 @@
-    additional 2 or 3 layout trials by default independently of the
-    ``layout_trials`` keyword argument's value. A trivial
-    layout and its reverse are included for all backends, just like the :class:`.DenseLayout`
+    additional 2 or 3 layout trials by default, independently from the
```
```suggestion
    additional two or three layout trials by default, independently from the
```

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
For some reason, I can't add comments until line 23 of this file.  Here are some comments on the earlier lines: 

I don't understand this whole thing:  
"When using QPY formats 10, 11, or 12 there is a dependency on the version
    of ``symengine`` installed in the payload for serialized
    :class:`.ParamerExpression` if there is mismatched version of the installed
    ``symengine`` package between the environment that generated the payload with
    :func:`.qpy.dump` and the installed version that is trying to load the payload
    with :func:`.qpy.load`."
    
Would one of these rewrites work? 
    
When using QPY formats 10, 11, or 12, if the version of the ``symengine`` package installed in the environment that generated the payload by using :func:`.qpy.dump` does not match the version of the ``symengine`` package that is trying to use :func:`.qpy.load` to load the payload, you will get an error. If you encounter this error, install the ``symengine`` package version from the error message to load the payload.
     
Or more simply:
     
When using QPY formats 10, 11, or 12, if the version of the ``symengine`` package installed in the environment that generated the payload does not match the version of the ``symengine`` package that is trying to load the payload, you will get an error.  (I'm not sure we need to tell them how to fix that error, but maybe.)

raynelfss replied:
I believe the original is a valid explanation, although a bit too technical so it can lead to confusion. Option 1 is a good rewrite in my opinion.

ElePT replied:
We could re-add the point about parameter expressions, because that's what essentially causes the issue. What about:

When using QPY formats 10, 11, or 12 with circuits that contain :class:`.ParamerExpression`s, if the version of the ``symengine`` package installed in the environment that generated the payload (:func:.qpy.dump) does not match the version of ``symengine`` installed in the enviroment where the payload is loaded (:func:.qpy.load), you will get an error. If you encounter this error, install the ``symengine package`` version from the error message before calling :func:.qpy.load.

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
"QPY format version >= 13 (or < 10) will not have this issue and it is recommended
    if you're serializing :class:`.ParameterExpression` objects as part of your
    circuit or any :class:`.ScheduleBlock` objects you use version 13 to avoid
    this issue in the future."  
    
Suggestion: 
"QPY format versions 13 or later (or prior to 10) will not have this issue. Therefore, if you're serializing :class:`.ParameterExpression` objects as part of your
    circuit or any :class:`.ScheduleBlock` objects, it is recommended that you use version 13 to avoid
    this issue in the future.

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
"The :func:`.qpy.dump` function will now emit format version 13 by default." -> "The :func:`.qpy.dump` function now emits format version 13 by default.

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
"...by default will only be compatible with Qiskit >= 1.3.0." -> "...by default are only compatible with Qiskit 1.3.0 or later."

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
```diff
@@ -23,11 +23,4 @@ upgrade_qpy:
     be compatible with Qiskit >= 1.3.0. If you need for the payload to be
```
```suggestion
    be compatible with Qiskit >= 1.3.0. If the payload needs to be
```

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
```diff
@@ -23,11 +23,4 @@ upgrade_qpy:
     be compatible with Qiskit >= 1.3.0. If you need for the payload to be
     loaded by a older version of Qiskit you can use the ``version`` flag on
```
```suggestion
    loaded by an earlier version of Qiskit, use the ``version`` flag on
```

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
```diff
@@ -23,11 +23,4 @@ upgrade_qpy:
     be compatible with Qiskit >= 1.3.0. If you need for the payload to be
     loaded by a older version of Qiskit you can use the ``version`` flag on
     :func:`.qpy.dump` to emit a version compatible with earlier releases of
```
```suggestion
    :func:`.qpy.dump` to emit the appropriate version.
```

beckykd commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
```diff
@@ -23,11 +23,4 @@ upgrade_qpy:
-#     needs to be worded so that it does not depend on any information only
-#     available in another section, such as the prelude. This may mean repeating
-#     some details.
+    Qiskit. You can refer to :ref:`qpy_compatibility` for more details on this.
```
```suggestion
    Refer to :ref:`qpy_compatibility` for more details.
```

beckykd commented on releasenotes/notes/1.3/assign-parameters-perf-regression-fc8c9db134b1763d.yaml:
```diff
@@ -2,8 +2,8 @@
-    call :meth:`~.QuantumCircuit.assign_parameters` only once, with all assignments at the same
-    time, as this reduces the proportion of time spent on input normalization and error-checking
-    overhead.
+    1.2.0 when calling the method in a tight loop, resulting in the binding only a small number of
```
```suggestion
    1.2.0 when calling the method in a tight loop, that caused only a small number of 
```

beckykd commented on releasenotes/notes/1.3/assign-parameters-perf-regression-fc8c9db134b1763d.yaml:
```diff
@@ -2,8 +2,8 @@
-    time, as this reduces the proportion of time spent on input normalization and error-checking
-    overhead.
+    1.2.0 when calling the method in a tight loop, resulting in the binding only a small number of
+    parameters out of a heavily parametric circuit on each iteration. If possible, it is still
```
```suggestion
    parameters from a heavily parametric circuit to be bound on each iteration. If possible, it is still
```

beckykd commented on releasenotes/notes/1.3/backend-sampler-v2-level1-dc13af460cd38454.yaml:
```diff
@@ -1,7 +1,7 @@
-    Support for level 1 data was added to :class:`~.BackendSamplerV2` as was
+    Support for level 1 data was added to :class:`~.BackendSamplerV2`, as was
     support for passing options through to the ``run()`` method of the wrapped
     :class:`~.BackendV2`. The run options can be specified using a
```
```suggestion
    :class:`~.BackendV2`. The run options can be specified by using a
```

beckykd commented on releasenotes/notes/1.3/backend-sampler-v2-level1-dc13af460cd38454.yaml:
```diff
@@ -1,7 +1,7 @@
+    Support for level 1 data was added to :class:`~.BackendSamplerV2`, as was
     support for passing options through to the ``run()`` method of the wrapped
     :class:`~.BackendV2`. The run options can be specified using a
     ``"run_options"`` entry inside of the ``options`` dicitonary passed to
```
```suggestion
    ``"run_options"`` entry inside the ``options`` dictionary passed to
```

beckykd commented on releasenotes/notes/1.3/backend-sampler-v2-level1-dc13af460cd38454.yaml:
Comments from lines I can't comment on: 

"dictionary mapping argument names to values for passing to the backend's
    ``run()`` method. When a ``"meas_level"`` option with a value of 1 is set
    in the run options, the results from the backend will be treated as level 1
    results rather as bit arrays (the level 2 format)."
    
Suggestion:

"dictionary that maps argument names to values to be passed to the backend's
    ``run()`` method. When a ``"meas_level = 1 "`` is set
    in the run options, the results from the backend will be treated as level 1
    results instead of bit arrays (the level 2 format).

beckykd commented on releasenotes/notes/1.3/basicsimulator-config-copy-5ecdfdf161e488f2.yaml:
```diff
@@ -2,6 +2,6 @@
-    rather than a ``dict_keys`` instance, matching the expected type and
-    allowing for configuration instance to be deep copied.
+    configuration instance returned by the ``configuration()`` method is
+    now a list rather than a ``dict_keys`` instance, matching the expected
```
```suggestion
    now a list instead of a ``dict_keys`` instance, matching the expected
```

beckykd commented on releasenotes/notes/1.3/binary-arithmetic-gates-6cd2b1c8112febe0.yaml:
" addition two" -> " addition of two"

beckykd commented on releasenotes/notes/1.3/clib-grover-op-cb032144e899ed0d.yaml:
```diff
@@ -4,7 +4,7 @@ features_circuits:
     Added :func:`.grover_operator` to construct a Grover operator circuit, used in e.g.
```
```suggestion
    Added :func:`.grover_operator` to construct a Grover operator circuit, used in 
```

beckykd commented on releasenotes/notes/1.3/clib-grover-op-cb032144e899ed0d.yaml:
```diff
@@ -4,7 +4,7 @@ features_circuits:
     Added :func:`.grover_operator` to construct a Grover operator circuit, used in e.g.
     Grover's algorithm and amplitude estimation/amplification. This function is similar to
```
```suggestion
    Grover's algorithm and amplitude estimation/amplification, for example. This function is similar to
```

beckykd commented on releasenotes/notes/1.3/clib-grover-op-cb032144e899ed0d.yaml:
```diff
@@ -4,7 +4,7 @@ features_circuits:
     Added :func:`.grover_operator` to construct a Grover operator circuit, used in e.g.
     Grover's algorithm and amplitude estimation/amplification. This function is similar to
     :class:`.GroverOperator`, but does not require choosing the implementation of the
```
```suggestion
    :class:`.GroverOperator`, but does not require you to choose the implementation of the
```

beckykd commented on releasenotes/notes/1.3/clib-grover-op-cb032144e899ed0d.yaml:
```diff
@@ -4,7 +4,7 @@ features_circuits:
     Grover's algorithm and amplitude estimation/amplification. This function is similar to
     :class:`.GroverOperator`, but does not require choosing the implementation of the
-    multi-controlled X gate a-priori and let's the compiler choose the optimal decomposition
+    multi-controlled X gate a-priori and lets the compiler choose the optimal decomposition
```
```suggestion
    multi-controlled X gate, but instead lets the compiler determine the optimal decomposition.
```

beckykd commented on releasenotes/notes/1.3/clib-grover-op-cb032144e899ed0d.yaml:
```diff
@@ -4,7 +4,7 @@ features_circuits:
     :class:`.GroverOperator`, but does not require choosing the implementation of the
-    multi-controlled X gate a-priori and let's the compiler choose the optimal decomposition
+    multi-controlled X gate a-priori and lets the compiler choose the optimal decomposition
     instead. In addition to this, it does not wrap the circuit into an opaque gate and is 
```
```suggestion
    Additionally, it does not wrap the circuit into an opaque gate and is 
```

beckykd commented on releasenotes/notes/1.3/clib-grover-op-cb032144e899ed0d.yaml:
```diff
@@ -4,7 +4,7 @@ features_circuits:
-    multi-controlled X gate a-priori and let's the compiler choose the optimal decomposition
+    multi-controlled X gate a-priori and lets the compiler choose the optimal decomposition
     instead. In addition to this, it does not wrap the circuit into an opaque gate and is 
     faster as less decompositions are required to transpile.
```
```suggestion
    faster because fewer decompositions are required for transpilation.
```

beckykd reviewed (approved):
Lots of good stuff! let me know if you have any questions!

raynelfss committed:
Apply suggestions from code review

Co-authored-by: Rebecca Dimock <66339736+beckykd@users.noreply.github.com>

Cryoris commented on releasenotes/notes/1.3/add-gates-to-collect-clifford-af88dd8f7a2a4bf9.yaml:
```diff
@@ -1,7 +1,7 @@
 features_transpiler:
   - |
-    Add an argument ``matrix_based`` to the :class:`.CollectCliffords()` transpiler pass.
+    Add argument ``matrix_based`` to the :class:`.CollectCliffords()` transpiler pass.
```
Just a comment for consistency: I think we mostly write in past-tense, right? I.e. `Fixed something` or `Added something` in case you'd like to change this 🙂

Cryoris replied:
I'm not sure on this one, but I thought this shouldn't include the brackets? Below there's some more occurences and in particular 
```
:class:`.RZGate(np.pi/2)`
```
which looks sus -- but it might be more likely that I just don't know about this option 🙂 

```suggestion
    Add argument ``matrix_based`` to the :class:`.CollectCliffords` transpiler pass.
```

Cryoris commented on releasenotes/notes/1.3/add-mcx-plugins-85e5b248692a36db.yaml:
```diff
@@ -17,7 +17,7 @@ features_transpiler:
+    As well as:
 
     * :class:`.MCXSynthesisDefault`, choosing the most efficient synthesis
       method based on the number of clean and dirty ancilla qubits available.
```
I can't comment there but on the line below there's a typo:

"As an example, consider ~~how~~ the transpilation of the following circuit"

Cryoris commented on releasenotes/notes/1.3/binary-arithmetic-gates-6cd2b1c8112febe0.yaml:
`AdderGate` should be `HalfAdderGate` -- we changed this naming in the PR and apparently forgot the reno 🙈 (there's two occurrences of this here)

Cryoris commented on releasenotes/notes/1.3/binary-arithmetic-gates-6cd2b1c8112febe0.yaml:
```diff
@@ -14,7 +14,7 @@ features_circuits:
   - |
     Added the :class:`.MultiplierGate` for multiplication of two :math:`n`-qubit registers, that is
     :math:`|a\rangle |b\rangle \mapsto |a\rangle |b\rangle |a \cdot b\rangle`.
```
This misses the target register (the gate's docstring has it, I just missed it here)
```suggestion
    :math:`|a\rangle |b\rangle |0\rangle \mapsto |a\rangle |b\rangle |a \cdot b\rangle`.
```

raynelfss committed:
Apply suggestions from code review

Co-authored-by: Julien Gacon <gaconju@gmail.com>

raynelfss committed:
Check for typos in any release notes.

Co-authored-by: Rebecca Dimock <66339736+beckykd@users.noreply.github.com>
Co-authored-by: Julien Gacon <gaconju@gmail.com>
Co-authored-by: Elena Peña Tapia <57907331+ElePT@users.noreply.github.com>

raynelfss committed:
Add missing release note for #13237

raynelfss commented:
I have checked all release notes to the best of my abilities now. @beckykd feel free to further review these. I should be posting the prelude release note sometime tomorrow.

raynelfss committed:
Move new missing release note to 1.3 folder

beckykd commented on releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:
```diff
@@ -1,8 +1,9 @@
-    The Qiskit Pulse package is being deprecated and will be removed in Qiskit 2.0.0. Pulse-level
-    access is currently only supported on a subset of Eagle devices and not supported on the Heron architecture. Furthermore, newer IBM Quantum architectures will not support pulse-level access.
+    The Qiskit Pulse package is now deprecated and will be removed in Qiskit 2.0.0. Pulse-level
+    access is currently only supported on a subset of Eagle devices and not supported on the Heron architecture.
```
```suggestion
    access is currently only supported on a subset of Eagle devices and is not supported on the Heron architecture.
```

beckykd commented on releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:
```diff
@@ -1,8 +1,9 @@
   - |
-    The Qiskit Pulse package is being deprecated and will be removed in Qiskit 2.0.0. Pulse-level
-    access is currently only supported on a subset of Eagle devices and not supported on the Heron architecture. Furthermore, newer IBM Quantum architectures will not support pulse-level access.
+    The Qiskit Pulse package is now deprecated and will be removed in Qiskit 2.0.0. Pulse-level
```
```suggestion
    The Qiskit Pulse package is now deprecated and will be removed in Qiskit 2.0.0. This is because pulse-level
```

beckykd commented on releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:
```diff
@@ -1,8 +1,9 @@
+    The Qiskit Pulse package is now deprecated and will be removed in Qiskit 2.0.0. Pulse-level
+    access is currently only supported on a subset of Eagle devices and not supported on the Heron architecture.
+    Furthermore, newer IBM Quantum architectures will not support pulse-level access.
     As a consequence, supporting Pulse as a first-class citizen frontend in the Qiskit SDK itself makes little
```
```suggestion
```

beckykd commented on releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:
```diff
@@ -1,8 +1,9 @@
+    access is currently only supported on a subset of Eagle devices and not supported on the Heron architecture.
+    Furthermore, newer IBM Quantum architectures will not support pulse-level access.
     As a consequence, supporting Pulse as a first-class citizen frontend in the Qiskit SDK itself makes little
     sense going forward. The deprecation includes all pulse code in :mod:`qiskit.pulse` as well as functionality
```
```suggestion
    The deprecation includes all pulse code in :mod:`qiskit.pulse` as well as functionality
```

beckykd commented on releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:
```diff
@@ -1,8 +1,9 @@
+    Furthermore, newer IBM Quantum architectures will not support pulse-level access.
     As a consequence, supporting Pulse as a first-class citizen frontend in the Qiskit SDK itself makes little
     sense going forward. The deprecation includes all pulse code in :mod:`qiskit.pulse` as well as functionality
     dependant or related to pulse such as pulse visualization, serialization and custom calibration support. For
```
```suggestion
    dependent on or related to pulse, such as pulse visualization, serialization, and custom calibration support. For
```

beckykd commented on releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:
```diff
@@ -35,7 +36,7 @@ deprecations_transpiler:
       * :meth:`~qiskit.transpiler.Target.instruction_schedule_map`
 
-    In addition the following transpiler passer are also being deprecated:
+    In addition the following transpiler passes are also being deprecated:
```
```suggestion
    In addition, the following transpiler passes are also being deprecated:
```

beckykd commented on releasenotes/notes/1.3/fix-estimator-reset-9e7539776df4cac4.yaml:
```diff
@@ -3,9 +3,9 @@ features_primitives:
-    The result was not reproducible, but it is now reproducible
-    if a random seed is set. For example::
+    a reset for some subsystems.
+    The result was previously not reproducible, but now it can be if a
```
```suggestion
    The result was previously not reproducible, but now it can be reproduced if a
```

beckykd commented on releasenotes/notes/1.3/fix-sparse-pauli-op-apply-layout-zero-43b9e70f0d1536a6.yaml:
```diff
@@ -1,7 +1,7 @@
   - |
     Fixed :meth:`.SparsePauliOp.apply_layout` to work correctly with zero-qubit operators.
-    For example, if you previously created a 0 qubit and applied a layout like::
+    For example, if you previously created a zero-qubit operator and applied a layout like::
```
```suggestion
    For example, previously if you created a zero-qubit operator and applied a layout like the following, it would have raised an error::
```

beckykd commented on releasenotes/notes/1.3/fix-sparse-pauli-op-apply-layout-zero-43b9e70f0d1536a6.yaml:
For line 9 (which I can't comment on)

"this would have previously raised an error. Now this will correctly return an operator of the form:" -> "Now, the above code correctly returns an operator of the following form::"

beckykd commented on releasenotes/notes/1.3/fixes_GenericBackendV2-668e40596e1f070d.yaml:
```diff
@@ -1,4 +1,7 @@
 fixes:
   - |
-    The constructor :class:`.GenericBackendV2` was allowing to create malformed backends because it accepted basis gates that couldn't be allocated in the backend size . That is, a backend with a single qubit should not accept a basis with two-qubit gates.
+    The constructor :class:`.GenericBackendV2` was allowing to create malformed
```
```suggestion
    The constructor :class:`.GenericBackendV2` was allowing you to create malformed
```

beckykd commented on releasenotes/notes/1.3/fixes_GenericBackendV2-668e40596e1f070d.yaml:
```diff
@@ -1,4 +1,7 @@
-    The constructor :class:`.GenericBackendV2` was allowing to create malformed backends because it accepted basis gates that couldn't be allocated in the backend size . That is, a backend with a single qubit should not accept a basis with two-qubit gates.
+    The constructor :class:`.GenericBackendV2` was allowing to create malformed
+    backends because it accepted basis gates that couldn't be allocated in the
+    backend size . That is, a backend with a single qubit should not accept a
```
```suggestion
    specified backend size. That is, a backend with a single qubit should not accept a
```

beckykd commented on releasenotes/notes/1.3/hls-with-ancillas-d6792b41dfcf4aac.yaml:
```diff
@@ -6,8 +6,8 @@ features_transpiler:
+    additional argument: ``qubits_initially_zero``. If set to ``True``, the pass assumes that the
     qubits are initially in the state :math:`|0\rangle`. In addition, the pass keeps track of
     clean and dirty auxiliary qubits throughout the run, and passes this information to plugins
     via kwargs ``num_clean_ancillas`` and ``num_dirty_ancillas``.
```
```suggestion
    by using kwargs ``num_clean_ancillas`` and ``num_dirty_ancillas``.
```

beckykd commented on releasenotes/notes/1.3/mcmt-gate-a201d516f05c7d56.yaml:
Line 9:

"Specific implementation can be chosen by specifying the high-level synthesis plugin::" -> "The desired implementation can be chosen by specifying the high-level synthesis plugin::"

beckykd commented on releasenotes/notes/1.3/mcmt-gate-a201d516f05c7d56.yaml:
```diff
@@ -21,8 +21,8 @@ features_circuits:
 
-    The :class:`.MCMTGate` in addition also supports custom (i.e., open) control states of
-    the control qubits.
+    Additionally, the :class:`.MCMTGate` also supports custom (i.e., open) control states
```
```suggestion
    Additionally, the :class:`.MCMTGate` also supports custom (that is, open) control states
```

beckykd commented on releasenotes/notes/1.3/outcome_bitstring_target_for_probabilities_dict-e53f524d115bbcfc.yaml:
```diff
@@ -3,11 +3,11 @@ features:
     The :class:`.StabilizerState` class now has a new method 
     :meth:`~.StabilizerState.probabilities_dict_from_bitstring` allowing the 
-    user to pass single bitstring to measure an outcome for. Previouslly the 
+    user to pass single bitstring to measure an outcome for. Previously the 
```
```suggestion
    user to pass a single bitstring to measure an outcome for. Previously, the 
```

beckykd commented on releasenotes/notes/1.3/outcome_bitstring_target_for_probabilities_dict-e53f524d115bbcfc.yaml:
```diff
@@ -3,11 +3,11 @@ features:
-    user to pass single bitstring to measure an outcome for. Previouslly the 
+    user to pass single bitstring to measure an outcome for. Previously the 
     :meth:`~.StabilizerState.probabilities_dict` would be utilized and would 
     at worst case calculate (2^n) number of probability calculations (depending
```
```suggestion
    in the worst case, calculate (2^n) probability calculations (depending
```

beckykd commented on releasenotes/notes/1.3/outcome_bitstring_target_for_probabilities_dict-e53f524d115bbcfc.yaml:
```diff
@@ -3,11 +3,11 @@ features:
     :meth:`~.StabilizerState.probabilities_dict_from_bitstring` allowing the 
-    user to pass single bitstring to measure an outcome for. Previouslly the 
+    user to pass single bitstring to measure an outcome for. Previously the 
     :meth:`~.StabilizerState.probabilities_dict` would be utilized and would 
```
```suggestion
    :meth:`~.StabilizerState.probabilities_dict` would be utilized and would, 
```

beckykd commented on releasenotes/notes/1.3/outcome_bitstring_target_for_probabilities_dict-e53f524d115bbcfc.yaml:
```diff
@@ -3,11 +3,11 @@ features:
     to measure the probability for. As the number of qubits increases, the more 
-    prevelant the performance enhancement may be (depending on the state) as only 
+    prevalent the performance enhancement may be (depending on the state) as only 
     1 bitstring result is measured.
```
```suggestion
    one bitstring result is measured.
```

beckykd commented on releasenotes/notes/1.3/oxidize-basis-translator-e246a99118bd9486.yaml:
```diff
@@ -0,0 +1,7 @@
+  - |
+    The :class:`.BasisTranslator` transpiler pass has been reimplemented
+    to work purely in Rust, taking advantage of all the other structures
+    that currently live there such as the :class:`.Target`, :class:`.EquivalenceLibrary`,
```
```suggestion
    that currently exist there, such as the :class:`.Target`, :class:`.EquivalenceLibrary`,
```

beckykd commented on releasenotes/notes/1.3/oxidize-basis-translator-e246a99118bd9486.yaml:
```diff
@@ -0,0 +1,7 @@
+    The :class:`.BasisTranslator` transpiler pass has been reimplemented
+    to work purely in Rust, taking advantage of all the other structures
+    that currently live there such as the :class:`.Target`, :class:`.EquivalenceLibrary`,
+    and the :class:`.DAGCircuit`. The pass itself now runs anywhere from
```
```suggestion
    and the :class:`.DAGCircuit`. The pass itself now runs 
```

beckykd commented on releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml:
"Added :meth:`.ProductFormula.expand` which allows to view the expansion of a product formula" -> "Added :meth:`.ProductFormula.expand`, which lets you view the expansion of a product formula"

beckykd commented on releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml:
"but faster as it
    internally uses Rust. The larger the circuit (e.g. by the Hamiltonian size, the number 
    of timesteps, or the Suzuki-Trotter order), the higher the speedup." -> "but faster, as it
    internally uses Rust. The larger the circuit (for example. by the Hamiltonian size, the number 
    of timesteps, or the Suzuki-Trotter order), the greater the speedup."

beckykd commented on releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml:
"``"upto_phase" = False`` allowing to obtain smaller" -> "``"upto_phase" = False``, allowing you to obtain smaller"

beckykd commented on releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml:
"For the full list of supported options, see" -> "For the full list of supported options, see the"

beckykd commented on releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml:
"to diagonalize the Pauli-Y
    operator: :class:`.PauliEvolutionGate`, :class:`.EvolvedOperatorAnsatz`, " -> "to diagonalize the Pauli-Y
    operator: :class:`.PauliEvolutionGate`, :class:`.EvolvedOperatorAnsatz`, and "

beckykd commented on releasenotes/notes/1.3/port-synth-cz-depth-line-mr-to-rust-1376d5a41948112a.yaml:
```diff
@@ -1,6 +1,11 @@
+    Port :func: `.synth_cz_depth_line_mr` to Rust. This function synthesizes a CZ
+    circuit for linear nearest neighbor (LNN) connectivity, based on the Maslov
+    and Roetteler method. On a 350x350 binary matrix, the Rust implementation
+    yields a speedup of about 30 times.
```
```suggestion
    yields a speedup of about 30x.
```

beckykd commented on releasenotes/notes/1.3/port-synth-cz-depth-line-mr-to-rust-1376d5a41948112a.yaml:
```diff
@@ -1,6 +1,11 @@
   - |
-    Port :func: `.synth_permutation_reverse_lnn_kms` to Rust, which synthesizes a reverse permutation for linear nearest-neighbor architecture using the Kutin, Moulton, Smithline method.
+    Port :func: `.synth_permutation_reverse_lnn_kms` to Rust, which synthesizes a
+    reverse permutation for linear nearest-neighbor architecture using the Kutin,
```
```suggestion
    reverse permutation for LNN architecture using the Kutin,
```

beckykd commented on releasenotes/notes/1.3/rust-commutation-checker-c738e67efa9d292f.yaml:
```diff
@@ -1,6 +1,6 @@
-    The the :class:`.CommutationChecker` class has been reimplemented in
+    The :class:`.CommutationChecker` class has been reimplemented in
     Rust. This retains the same functionality as before but is now
     significantly in most cases.
```
```suggestion
    significantly faster in most cases.
```

beckykd commented on releasenotes/notes/1.3/rust-consolidation-a791a00380fc78b8.yaml:
```diff
@@ -1,7 +1,7 @@
-    The :class:`.ConsolidateGates` pass will now run the equivalent of the
+    The :class:`.ConsolidateBlocks` pass will now run the equivalent of the
     :class:`.Collect2qBlocks` pass internally if it was not run in a pass
     manager prior to the pass. Previously it was required that
```
```suggestion
    manager prior to the pass. Previously, 
```

beckykd commented on releasenotes/notes/1.3/rust-consolidation-a791a00380fc78b8.yaml:
```diff
@@ -1,7 +1,7 @@
+    The :class:`.ConsolidateBlocks` pass will now run the equivalent of the
     :class:`.Collect2qBlocks` pass internally if it was not run in a pass
     manager prior to the pass. Previously it was required that
     :class:`.Collect2qBlocks` or :class:`.Collect1qRuns` were run prior to
```
```suggestion
    :class:`.Collect2qBlocks` or :class:`.Collect1qRuns` had to be run before
```

beckykd commented on releasenotes/notes/1.3/rust-consolidation-a791a00380fc78b8.yaml:
"By doing the collection internally the overhead of the pass
    is reduced. " -> "By doing the collection internally, the overhead from the pass
    is reduced. "

beckykd commented on releasenotes/notes/1.3/rust-consolidation-a791a00380fc78b8.yaml:
":class:`.ConsolidateBlocks` the collected" -> ":class:`.ConsolidateBlocks`, the collected"

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
In several places, there is only one colon introducing a list. Should there be two? for example "The new methods are:"

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
"is added to manually" -> "was added to manually"

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
I'm not sure how to fix this one: "has two need keyword arguments, ``var_remap`` and ``inline_captures``"

Maybe "has two required keyword arguments, ``var_remap`` and ``inline_captures``"

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
I'm not sure what the term "inlined" means.  Can you rephrase?
"inlined onto the base circuit"

raynelfss replied:
I believe it's just mentioning how it's realigning new instructions onto a base circuit. Maybe the word re-align works better?

jakelishman replied:
"Inline" is a compiler term

jakelishman replied:
But also I think that note is from some prior release?

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
I'm not sure what this is saying.  I suggested a rewrite, but I might be changing the meaning:

"In conjunction with the ``vars_mode="captures"`` option to several
    :class:`.DAGCircuit` methods, this can be used to combine DAGs that operate on the same variables." -> "Along with the ``vars_mode="captures"`` option available on several
    :class:`.DAGCircuit` methods, this can be used to combine DAGs that operate on the same variables."

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
I'm also not sure what this means.  Maybe my suggestion is correct or maybe it would be more clear to rephrase it like "If set to "true", this ..(does some thing)." so it's active voice.

 "which can be set to
    ``True`` to inline "captured" :class:`.Var` nodes on the argument circuit onto the base circuit
    without redeclaring them.  In conjunction with the ``vars_mode="captures"`` option to several
    :class:`.DAGCircuit` methods, this can be used to combine DAGs that operate on the same variables."
    
    Possible rewrite:
    
     which can be set to
    ``True`` to inline "capture" :class:`.Var` nodes on the argument circuit onto the base circuit
    without redeclaring them.  Combined with the ``vars_mode="captures"`` option available on several
    :class:`.DAGCircuit` methods, this can be used to combine DAGs that operate on the same variables.

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
"for more detail on the format changes." -> "for more details on the format changes."

beckykd commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
"Implementations of :class:`.BackendV2` (and :class:`.BackendV1`) may desire to update their 
    :meth:`~.BackendV2.run` methods to eagerly reject inputs containing typed
    classical variables (see :mod:`qiskit.circuit.classical`) and the :class:`.Store` instruction,
    if they do not have support for them." - > "Implementations of :class:`.BackendV2` (and :class:`.BackendV1`) should update their 
    :meth:`~.BackendV2.run` methods to reject inputs that contain typed
    classical variables (see :mod:`qiskit.circuit.classical`) and the :class:`.Store` instruction
    if such input is not supported."

beckykd reviewed (commented):
I think I'm through everything!  Thanks for putting them all together and doing that initial edit!

raynelfss committed:
Apply suggestions from code review

Co-authored-by: Rebecca Dimock <66339736+beckykd@users.noreply.github.com>

raynelfss committed:
Apply code suggestions from code review
Co-authored-by: Rebecca Dimock <66339736+beckykd@users.noreply.github.com>
Co-authored-by: Julien Gacon <gaconju@gmail.com>

raynelfss committed:
More code review comments

Co-authored-by: Rebecca Dimock <66339736+beckykd@users.noreply.github.com>

raynelfss committed:
Add prelude note

raynelfss committed:
Fix: Incorrent indentation of prelude release note.

raynelfss committed:
Fix: Spelling errors and broken links.

raynelfss committed:
Move release note to 1.3 folder

ElePT commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+
+      * Improvements to the circuit library that improve compilation quality and
+        construction speed. These changes include improvements such as improved qubit
+        tracking for :class:`.HighLevelSynthesis`, structural circuits as part of the library
```
While we definitely want to highlight improvements, maybe having 4 instances of the word in two lines is a bit two much? :) What about:

```suggestion
      * Improvements to the circuit library that increase compilation quality and
        construction speed. These changes include developments such as improved qubit
        tracking for :class:`.HighLevelSynthesis`, structural circuits as part of the library
```

raynelfss committed:
Update prepare-1.3.0-7c45598775fc2bbf.yaml

Co-authored-by: Elena Peña Tapia <57907331+ElePT@users.noreply.github.com>

mtreinish commented on releasenotes/notes/1.3/storage-var-a00a33fcf9a71f3f.yaml:
```diff
@@ -6,7 +6,7 @@ features_circuits:
 
     See :ref:`circuit-real-time-methods` for more in-depth discussion on all of these.
```
This note file needs to be deleted it was already released in 1.1.0.

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+---
+prelude: >
+    Qiskit version 1.3.0 brings in major performance and quality improvements
+    for the transpiler. The minimum required Python version is now 3.9.
```
What are the quality improvements?

raynelfss replied:
With this, I was mainly referring to the refactoring of the circuit library and the many additions to the synthesis transpiler passes and plugins, all of which were aimed at better performance and quality of the transpiler. But I can change this if you believe it is not the best way of putting it.

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+---
+prelude: >
+    Qiskit version 1.3.0 brings in major performance and quality improvements
+    for the transpiler. The minimum required Python version is now 3.9.
```
I'd probably move this to the end since it's more just a secondary FYI and not a headline thing.

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+    for the transpiler. The minimum required Python version is now 3.9.
+
+    There have been many new features, fixes and improvements introduced in this
+    new version of Qiskit, some of which include:
```
I'd say highlights or something to else to put more importance on this. These are the top features in this release.

```suggestion
    new version of Qiskit, the highlights are:
```

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+        `Rustiq <https://github.com/smartiel/rustiq>`_.
+
+      * The deprecation of the Qiskit pulse module and all of its related components. 
+        The pulse module and its functionality will be part of `Qiskit Dynamics <https://github.com/qiskit-community/qiskit-dynamics>`_.
```
Typically I wouldn't highlight this in the prelude. We have the full release notes and a dedicated deprecation section for this. The prelude is really just for the most important changes in the release and a deprecation normally doesn't fall into this in my opinion. The removal in 2.0 probably would though (although those release notes are going to be huge).

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+        manager have been ported into Rust, resulting in a 6x overall runtime 
+        improvement compared to Qiskit 1.2.4. The ported passes include :class:`.BasisTranslator`,
+        :class:`.CommutationAnalysis`, :class:`.ConsolidateBlocks`, :class:`.UnitarySynthesis`,
+        and others.
```
I would call out the dag circuit, target, and other data model pieces being in rust too. This is much more than just these 4 passes although they were the primary bottlenecks, there was a lot more too it than this sounds like.

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+        construction speed. These changes include developments such as improved qubit
+        tracking for :class:`.HighLevelSynthesis`, structural circuits as part of the library
+        such as IQP, and the addition of :class:`.PauliEvolutionGate` and its operability with 
+        `Rustiq <https://github.com/smartiel/rustiq>`_.
```
This really needs to be two release bullet points. It's conflating the circuit library refactor and the introduction of a high level synthesis plugin for the Pauli evolution gate. These should be separate bullet points. Also The pauli evolution gate class already existed the work done in this release was about refactoring that in rust, and starting to leverage the high level synthesis interface for doing the synthesis.

Maybe @Cryoris or @alexanderivrii can help you write a couple of sentences about this.

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+      * Improvements to the circuit library that increase compilation quality and
+        construction speed. These changes include developments such as improved qubit
+        tracking for :class:`.HighLevelSynthesis`, structural circuits as part of the library
+        such as IQP, and the addition of :class:`.PauliEvolutionGate` and its operability with 
```
```suggestion
        such as :func:`.iqp`. Additionally the :class:`.PauliEvolutionGate` has been updated to leverage high level synthesis plugins including a new plugin that leverages the circuit synthesis library 
```

I would be tempted to add a sphinx ref to the docs section on high level synthesis plugins so there is a link to the details.

mtreinish commented on releasenotes/notes/1.3/paulifeaturemap-takes-dictionary-as-entanglement-02037cb2d46e1c41.yaml:
```diff
@@ -1,9 +1,9 @@
     could be given as ``List[int]`` or ``List[List[int]]``, which was incompatible with the fact
-    that entanglement blocks of different sizes are used. Instead, the entanglement can be 
+    that entanglement blocks of different sizes are used. Now, the entanglement can be 
     given as dictionary with ``{block_size: entanglement}`` pairs.
```
Is this actually a fix? This reads like a feature note (which there is one below for) about the addition of support for dictionary inputs. What is being fixed here and can we get a concise description of what the bug was before 1.3.0 and how it's fixed? If the answer is it's actually just that the list input was not covering all use cases I would delete this note and rely on just the feature note below.

raynelfss replied:
I'm okay with deleting the fix section, since it already mentions the specification of entanglement with a dictionary down below. I will remove it if @Cryoris agrees.

Cryoris replied:
Let's delete this -- 1.3 doesn't "fix" the old path of giving the entanglement as `List[List[int]]`, the old path is just invalid and will still fail.

mtreinish commented on releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml:
```diff
@@ -45,14 +45,14 @@ features_synthesis:
+    operator: :class:`.PauliEvolutionGate`, :class:`.EvolvedOperatorAnsatz`, and  
     :class:`.PauliFeatureMap`. Previously, these classes used either :math:`H S` or
     :math:`R_X(-\pi/2)` as basis transformation. Using the :math:`\sqrt{X}` operation, 
     represented by the :class:`.SXGate` is more efficient as it uses only a single gate
```
Is there an alternative for people if they were relying on the previous gates to diagonalize the operator? Normally for an upgrade note we would explain how a user could retain the behavior if they needed for some reason. A valid solution would be to generate the circuit in 1.2.x and export it with QPY. But it's helpful for users to have whatever the alternative path is called out explicitly.

Cryoris replied:
There's no flag to turn the old behavior on, only to either write a custom transformation pass or what you suggested 🤔

mtreinish commented on releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml:
```diff
@@ -15,18 +15,18 @@ features_quantum_info:
   - |
-    Added :meth:`.ProductFormula.expand` which allows to view the expansion of a product formula
+    Added :meth:`.ProductFormula.expand`, which lets you view the expansion of a product formula
     in a sparse Pauli format.
```
Do you think an example here would be useful?

Cryoris replied:
```suggestion
    Added :meth:`.ProductFormula.expand`, which lets you view the expansion of a product formula
    in a sparse Pauli format. For example, we can query the format of a second-order Trotter expansion
    of a Hamiltonian as::
    
        from qiskit.quantum_info import SparsePauliOp
        from qiskit.circuit.library import PauliEvolutionGate
        from qiskit.synthesis.evolution import SuzukiTrotter
        
        hamiltonian = SparsePauliOp(["IX", "XI", "ZZ"], coeffs=[-1, -1, 1])
        evo = PauliEvolutionGate(hamiltonian, time=3.14)
        trotter = SuzukiTrotter(order=2)
        print(trotter.expand(evo))
        
    which will print
    
    .. parsed-literal::
    
        [('X', [0], -3.14), ('X', [1], -3.14), ('ZZ', [0, 1], 6.28), 
        ('X', [1], -3.14), ('X', [0], -3.14)]

```

mtreinish commented on releasenotes/notes/1.3/outcome_bitstring_target_for_probabilities_dict-e53f524d115bbcfc.yaml:
```diff
@@ -3,11 +3,11 @@ features:
   - |
```
This release note should be in `features_quantum_info`

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,22 @@
+        :class:`.CommutationAnalysis`, :class:`.ConsolidateBlocks`, :class:`.UnitarySynthesis`,
+        and others.
+
+      * Improvements to the circuit library that increase compilation quality and
```
Do we want to call out ancilla allocation in hls here?

raynelfss replied:
How can I better write something for this? @alexanderivrii

alexanderivrii replied:
Just yesterday @Cryoris and I were writing a paragraph for Luciano's blog post, so trying to shorten what we wrote and combine with what you already have, I can suggest writing something like this:

```
* Improvements to the circuit library that increase compilation quality and
  circuit construction speed. Structural operations that have a unique
  decomposition are represented as functions that return a :class:`.QuantumCircuit`. 
  New examples include :func:`.quantum_volume`, :func:`.qaoa_ansatz`, and 
  :func:`.efficient_su2`. Most of these functions are built in Rust, resulting in 
  significantly faster circuit construction runtime.
  The abstract operations that can be implemented in multiple different ways
  in terms of the lower-level gates are represented as :class:`.Gate` or 
  :class:`.Instruction`. New examples include :class:`.ModularAdderGate`, 
  :class:`.MCMTGate`, and :class:`.GraphStateGate`. This allows building a quantum 
  circuit without choosing a specific implementation for these gates, and letting 
  the compiler choose the optimal decomposition. This is especially powerful
  in combination with the improvements to the :class:`.HighLevelSynthesis`
  transpiler pass that tries to find the highest-quality decomposition available
  for a given gate, and can now leverage idle auxiliary qubits. There is a large
  number of new available plugins, including a new plugin for 
  :class:`.PauliEvolutionGate` that leverages the circuit synthesis library 
  `Rustiq <https://github.com/smartiel/rustiq>`_.
```
But you may want to shorten this a bit further and improve the wording.

Cryoris replied:
Extending on Sasha's suggestion:
```
* Improvements to the circuit library that increase compilation quality and
  circuit construction speed. Operations are now distinguished into:
  
   * Structural operations, that have a unique decomposition, and are 
      represented as functions that return a :class:`.QuantumCircuit`
      (for example: :func:`.real_amplitudes`). 
      Most of these functions are built in Rust, resulting in 
      significantly faster circuit construction runtime.
   * Abstract operations, that can be implemented using different 
      decompositions, are represented as :class:`.Gate` or :class:`.Instruction`
      (for example: :class:`.PauliEvolutionGate`).
      This allows building an abstract quantum circuit and letting 
      the compiler choose the optimal decomposition.
      
  Using an abstract circuit description is especially powerful in combination 
  with improvements to the :class:`.HighLevelSynthesis` transpiler pass,
  which can now take into account idle auxiliary qubits to find the best available 
  decomposition for a given gate.
```

mtreinish commented on releasenotes/notes/1.3/hls-with-ancillas-d6792b41dfcf4aac.yaml:
```diff
@@ -6,11 +6,11 @@ features_transpiler:
+    by using kwargs ``num_clean_ancillas`` and ``num_dirty_ancillas``.
 upgrade_transpiler:
   - |
     The :func:`~qiskit.compiler.transpile` now assumes that the qubits are initially in the state
```
I think some explanation around this change or why this could be undesirable would be good.

alexanderivrii replied:
Per @mtreinish's comment,  it might be worth to warn the users that the default value of ``qubits_initially_zero`` is ``True``, and there were cases in the past were this assumption was problematic  (see #6943).

Maybe we could change the first paragraph to something like this:

```
A new argument ``qubits_initially_zero`` has been added to :func:`qiskit.compiler.transpile`,
:func:`.generate_preset_pass_manager`, and to :class:`~.PassManagerConfig`.
If set to the default value of ``True``, the qubits are assumed to be initially in the state 
:math:`|0\rangle`, potentially allowing additional optimization opportunities for individual 
transpiler passes. In particular, the :class:`.HighLevelSynthesis` transpiler pass will choose 
a better decomposition for each :class:`.MCXGate` gate in a circuit when an idle auxiliary qubit 
in the state :math:`|0\rangle` is available. 

However, there are cases when ``qubits_initially_zero`` should be set to ``False``, such 
as when transpiling for backends that do not properly initialize qubits, or when manually
calling transpiler passes on subcircuits of a larger quantum circuit. 
```

mtreinish commented on releasenotes/notes/1.3/dag-oxide-60b3d7219cb21703.yaml:
```diff
@@ -1,25 +1,23 @@
+    should be fully API compatible with the previous Python implementation. While the class
+    was previously implemented using `rustworkx <https://www.rustworkx.org/>`__, for which
+    the underlying data graph structure exists in Rust, the implementation of the class and
+    all the data was lived in Python. This new version of :class:`.DAGCircuit` stores Rust friendly
```
What does "Rust friendly" mean here? The data types are defined in rust instead of python I think the "Rust native representation" in previous version was more descriptive.

raynelfss replied:
Yes, I changed the wording here a little because the DAGCircuit still keeps some structures as PyO3-wrapped Python structures so it doesn't have rust native representations for all its data. However, you are right that most of it is, in fact, using Rust native structures, and I should change the wording here.

mtreinish commented on releasenotes/notes/1.3/dag-oxide-60b3d7219cb21703.yaml:
```diff
@@ -1,25 +1,23 @@
+    all the data was lived in Python. This new version of :class:`.DAGCircuit` stores Rust friendly
+    datatypes and is more memory-efficient due to the compressed qubit and clbit representation
+    designed for instructions at rest. It also enables transpiler passes to fully manipulate a
+    :class:`.DAGCircuit` from Rust, enabling potential improvements in performance.
```
We can drop "potential" here because we have fulfilled the potential now. When I wrote this originally it was speculative because we hadn't started leveraging this yet. But now we have.

```suggestion
    :class:`.DAGCircuit` from Rust, enabling improvements in performance.
```

mtreinish commented on releasenotes/notes/1.3/backend-sampler-v2-level1-dc13af460cd38454.yaml:
```diff
@@ -1,15 +1,15 @@
 ---
 features:
```
This should be `features_primitives`

```suggestion
features_primitives:
```

mtreinish commented on releasenotes/notes/1.3/backend-sampler-v2-level1-dc13af460cd38454.yaml:
```diff
@@ -1,15 +1,15 @@
     in the run options, the results from the backend will be treated as level 1
-    results rather as bit arrays (the level 2 format).
+    results instead of bit arrays (the level 2 format).
 upgrade:
```
Is this really an upgrade note of a fix note? If it's an upgrade note we should explain the rationale behind the change, and also change the category to `upgrade_primitives`.

mtreinish commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
```diff
@@ -5,29 +5,22 @@ features_qpy:
+    not match the version of ``symengine`` installed in the enviroment where the
+    payload is loaded (:func:.qpy.load), you will get an error. If you encounter
+    this error, install the ``symengine`` version from the error message
+    before calling :func:.qpy.load. QPY format versions 13 or later
```
```suggestion
    before calling :func:`.qpy.load`. QPY format versions 13 or later
```

mtreinish commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
```diff
@@ -5,29 +5,22 @@ features_qpy:
+    :class:`.ParamerExpression`s, if the version of the ``symengine`` package
+    installed in the environment that generated the payload (:func:.qpy.dump) does
+    not match the version of ``symengine`` installed in the enviroment where the
+    payload is loaded (:func:.qpy.load), you will get an error. If you encounter
```
```suggestion
    payload is loaded (:func:`.qpy.load`), you will get an error. If you encounter
```

mtreinish commented on releasenotes/notes/1.3/add-qpy-v13-3b22ae33045af6c1.yaml:
```diff
@@ -5,29 +5,22 @@ features_qpy:
-    circuit or any :class:`.ScheduleBlock` objects you use version 13 to avoid
+    When using QPY formats 10, 11, or 12 with circuits that contain
+    :class:`.ParamerExpression`s, if the version of the ``symengine`` package
+    installed in the environment that generated the payload (:func:.qpy.dump) does
```
```suggestion
    installed in the environment that generated the payload (:func:`.qpy.dump`) does
```

mtreinish reviewed (changes requested):
Thanks for doing this, I've done a quick first pass through the notes but I still need to read the rendered output in sphinx to see how the complete notes read. From this first pass though I have a few concerns before I feel this is ready to merge and subsequently tagging the release.

The first is there are some notes that are already published on prior releases we need to delete those files from the `stable/1.3` branch in this PR to avoid duplicating them. You can just `git rm` these files, but please do an audit of all the notes in the rendered output to make sure none exist on https://docs.quantum.ibm.com/api/qiskit/release-notes/1.2 or https://docs.quantum.ibm.com/api/qiskit/release-notes/1.1

The second large concern is around how we're documenting performance improvements. We are very inconsistent about how we're reporting speedups across the board. A lot of synthesis functions are called out individually as having been ported to rust and citing a speedup factor. While for the transpiler passes only some do this, the basis translator stands out prominently as being one of the few that did this. If we're going to call these out as feature notes we should be consistent and ensure we do it for all of them. Or if we feel it's not important to document them individually as a long list it is probably better to consolidate the multiple notes into a single note for passes that lists out the passes we've ported to rust and they're generally faster. We should do the same thing for synthesis functions too. Personally I think a single large note documenting all of them is better for readers than doing a lot of small notes.

A smaller thing is I feel like many release notes were not checked over for completeness or correct categorization and we should ensure this is all correct before publishing the release notes.

raynelfss committed:
Move release notes from 1.1, 1.2 into their respective folders.

raynelfss committed:
Apply suggestions from code review

Co-authored-by: Matthew Treinish <mtreinish@kortar.org>

raynelfss committed:
Apply suggestions from code review

Co-authored-by: Matthew Treinish <mtreinish@kortar.org>

raynelfss committed:
More review comments

Co-authored-by: Matthew Treinish <mtreinish@kortar.org>

raynelfss committed:
Move latest release note

raynelfss committed:
Fix a couple of sphinx formatting errors

raynelfss committed:
Change wording on prelude

ElePT commented on releasenotes/notes/1.3/deprecate-basic-simulator-configuration-9d782925196993e9.yaml:
```diff
@@ -8,7 +8,7 @@ deprecations_providers:
 
     Here is a quick guide for accessing the most common ``BackendConfiguration`` attributes in the 
-    :class:`BackendV2` model:""
+    :class:`BackendV2` model:
```
This should format the table as a code snippet and respect the structure:

```suggestion
    :class:`BackendV2` model::
```

ElePT committed:
Update releasenotes/notes/1.3/deprecate-basic-simulator-configuration-9d782925196993e9.yaml

raynelfss committed:
Update releasenotes/notes/1.3/pauli-evo-plugins-612850146c3f7d49.yaml

Co-authored-by: Julien Gacon <gaconju@gmail.com>

raynelfss committed:
More corrections on release notes

Co-authored-by: Julien Gacon <gaconju@gmail.com>
Co-authored-by: Alexander Ivrii <alexi@il.ibm.com>

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,35 @@
+        improvements to the :class:`.HighLevelSynthesis` transpiler pass, which can now take
+        into account idle auxiliary qubits to find the best available decomposition for a given gate.
+
+      * The minimum required Python version is now 3.9.
```
I'd probably write this as:

```suggestion
      * The minimum supported Python version is now 3.9 as Python 3.8 went end of life in 2024-10. Official support for Python 3.13 support was also added in this release.
```

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,35 @@
+      * Essential features of Qiskit have either been ported into Rust
+        or have a representation with Rust-native datatypes. Some of these
+        components include the :class:`.DAGCircuit`, the :class:`.Target`,
+        the :class:`.EquivalenceLibrary`, and others.
```
Core 
```suggestion
      * The core data structures for transpilation in Qiskit have been ported
         to Rust internally. This includes components such as the
         :class:`.DAGCircuit`, :class:`.Target`, :class:`.EquivalenceLibrary`,
        and others. The public APIs for all of these data structures remains
        unchanged but the performance after the rewrites has improved.
```

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,35 @@
+      * The majority of the transpiler passes used by the preset pass 
+        manager have been ported into Rust, resulting in a 6x overall runtime 
+        improvement compared to Qiskit 1.2.4. The ported passes include :class:`.BasisTranslator`,
+        :class:`.CommutationAnalysis`, :class:`.ConsolidateBlocks`, and :class:`.UnitarySynthesis`.
```
```suggestion
      * The majority of the transpiler passes used by the preset pass 
        manager have been ported into Rust, resulting in an average 6x overall runtime 
        improvement compared to Qiskit 1.2.4 when running the
        `benchpress benchmarks <https://github.com/Qiskit/benchpress/>`__.
        Some of the particularly impactful passes that greatly benefited from
        the rewrites were the :class:`.BasisTranslator`,
        :class:`.CommutationAnalysis`, :class:`.ConsolidateBlocks`, and :class:`.UnitarySynthesis`.
        You can refer to the feature release notes for a full list of the ported passes.
```

mtreinish reviewed (commented):
Thanks for updating things. I started with the updated prelude, it is looking much better. Just have some small wording suggestions or missing details in suggestions

raynelfss committed:
Apply suggestions to prelude note

Co-authored-by: Matthew Treinish <mtreinish@kortar.org>

mtreinish commented on releasenotes/notes/1.3/prepare-1.3.0-7c45598775fc2bbf.yaml:
```diff
@@ -0,0 +1,40 @@
+        improvements to the :class:`.HighLevelSynthesis` transpiler pass, which can now take
+        into account idle auxiliary qubits to find the best available decomposition for a given gate.
+
+      * The minimum supported Python version is now 3.9 as Python 3.8 went end of life in 2024-10. Official support for Python 3.13 support was also added in this release.
```
This is missing a newline, probably from the web editor I used for my previous suggestions. Manually adding one line like this should fix the build error.

```suggestion
      * The minimum supported Python version is now 3.9 as Python 3.8 went end of life in 2024-10. Official support for Python 3.13 support was also added in this release.

```

raynelfss committed:
Consolidate transpiler pass port related release notes

raynelfss committed:
Move in new release note

raynelfss committed:
Modify new release note

mtreinish commented on releasenotes/notes/1.1/fix-apply-layout-duplicate-negative-indices-cf5517921fe52706.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.1/fix-swap-router-layout-f28cf0a2de7976a8.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.1/fix-symbolic-unit-scaling-c3eb4d9be674dfd6.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.1/parallel-check-8186a8f074774a1f.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.1/plot-circuit-layout-5935646107893c12.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.1/qasm2-bigint-8eff42acb67903e6.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.1/storage-var-a00a33fcf9a71f3f.yaml:
```diff
@@ -6,7 +6,7 @@ features_circuits:
     See :ref:`circuit-real-time-methods` for more in-depth discussion on all of these.
 
-    The new methods are:
+    The new methods are::
```
I don't think this is correct. The double `::` is used to start a literal block: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks instead you wanted a bulleted list (https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks). That being said this note file needs to be delete it's from 1.1.

mtreinish commented on releasenotes/notes/1.2/add-random-clifford-util-5358041208729988.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.2/backend-estimator-v2-variance-905c953415ad0e29.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.2/barebone-backend-option-675c86df4382a443.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.2/circuit-draw-warn-justify-03434d30cccda452.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.2/extended-random-circuits-049b67cce39003f4.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.2/fix-qft-plugins-7106029d33c44b96.yaml:
Just a reminder to delete this file.

mtreinish commented on releasenotes/notes/1.2/outcome_bitstring_target_for_probabilities_dict-e53f524d115bbcfc.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.2/product-formula-improvements-1bc40650151cf107.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.2/uniform-superposition-gate-3bd95ffdf05ef18c.yaml:
Just a reminder to delete this file instead of moving it.

mtreinish commented on releasenotes/notes/1.3/add-mcx-plugins-85e5b248692a36db.yaml:
```diff
@@ -17,12 +17,12 @@ features_transpiler:
       method based on the number of clean and dirty ancilla qubits available.
 
-    As an example, consider how the transpilation of the following circuit::
+    As an example, consider the transpilation of the following circuit::
```
This might be a better choice to use the plot directive and change the last line of the code example to `transpile(qc).draw("mpl")`

mtreinish commented on releasenotes/notes/1.3/backend-sampler-v2-level1-dc13af460cd38454.yaml:
```diff
@@ -1,16 +1,16 @@
-    results rather as bit arrays (the level 2 format).
-upgrade:
+    results instead of bit arrays (the level 2 format).
+upgrade_primitives:
```
For this note I would add a sentence to the end of this like:
```
If you were previously relying on this behavior you can manually
clear the metadata before calling :meth:`.BackendSamplerV2.run`
by calling `circuit.metadata.clear()`.
```

mtreinish commented on releasenotes/notes/1.3/clib-grover-op-cb032144e899ed0d.yaml:
```diff
@@ -1,12 +1,12 @@
+    Additionally, it does not wrap the circuit into an opaque gate and is 
+    faster because fewer decompositions are required for transpilation.
 
     Example::
```
You can use a plot directive here so the circuit is actually shown in the release note. You just need to change the `print(grover_op.draw())` to `grover_op.draw("mpl")`

mtreinish commented on releasenotes/notes/1.3/deprecate-basic-simulator-configuration-9d782925196993e9.yaml:
```diff
@@ -25,8 +25,8 @@ deprecations_providers:
       backend.configuration().max_shots          No representation      
 
-    (*) Note that ``backend.target.operation_names`` includes ``basis_gates`` and additional 
+    (*) Note that :meth:`.Backend.target.operation_names` includes ``basis_gates`` and additional 
```
This isn't correct and won't resolve correctly. It would be ``:attr:` instead of meth because target is an attribute of backend. But sphinx won't be able to resolve this because target isn't a path it can find as it's an instance attribute. This was probably more correct before.

mtreinish commented on releasenotes/notes/1.3/oxidize-transpiler-passes-16f2f1c57612e74a.yaml:
```diff
@@ -0,0 +1,53 @@
+---
+features_transpiler:
+  - |
```
Thanks for consolidating these into a single note. I think you just need to unify the language here. We should just list out the passes that were ported and say they're faster in general. We don't need any of the specific details for each pass.

mtreinish commented on releasenotes/notes/1.3/oxidize-transpiler-passes-16f2f1c57612e74a.yaml:
```diff
@@ -0,0 +1,53 @@
+      is reduced. If :class:`.Collect2qBlocks` or :class:`.Collect1qRuns` are
+      run prior to :class:`.ConsolidateBlocks,` the collected runs by those
+      passes from the property set are used and there is no change in behavior
+      for the pass. The pass now runs up to 3x faster than it did with Python.
```
Except for the last sentence this should be a separate note like it was before. This was documenting the new feature of the pass to run standalone without any of the collection passes.

raynelfss committed:
Add missing transpiler passes to the Rust epic
- Restored some deleted release notes.
- Fix formatting issues.

raynelfss committed:
Remove stray release notes

raynelfss committed:
Fix: Use plot directive effectively

mtreinish committed:
Remove all 1.2 release notes and build from 1.3.0b1

Due to how we end up branching 1.2.0 without a tag this is confusing
reno's version detection because on the stable/1.3 branch there is no
tag before 1.3.0b1 associated with any of the release notes for 1.2.0.
This commit just removes all the release notes to fix this.

Additionally, this corrects the earliest-version field in the sphinx
directive for reno so we include the notes associated with 1.3.0b1.

mtreinish committed:
Update transpiler pass porting note

mtreinish committed:
Copy edit most release notes

mtreinish committed:
Fix typos

mtreinish reviewed (dismissed):
I think this looks good now, while probably not perfect (no release's documentation is ever perfect) it's definitely looking really good to me now.. Thanks for all the work on polishing it and updating all the notes and also shepherding the 1.3.0 release.

raynelfss committed:
Slight reword in oxidize-transppiler-passes release note
